jenkins/Jenkinsfile

```diff
@@ -1,203 +1,167 @@
-      }
-    }
+  // build the war file (the binary).  This is the only
+  // place that happens.
+  stage('Build') {
+    sh './gradlew clean assemble'
+  }
 
-    // These are the Behavior Driven Development (BDD) tests
-    // See the files in src/bdd_test
-    // These tests do not require a running system.
-    stage('Automated Acceptance Tests') {
-      steps {
-        sh './gradlew generateCucumberReports'
-        // generate the code coverage report for jacoco
-        sh './gradlew jacocoTestReport'
-      }
-      post {
-          always {
-            junit 'build/test-results/bdd/*.xml'
-          }
-        }
-    }
+  // run all the unit tests - these do not require anything else
+  // to be running and most run very quickly.
+  stage('Unit Tests') {
+    sh(script: './gradlew test')
+    junit 'build/test-results/test/*.xml'
+  }
 
-    // Runs an analysis of the code, looking for any
-    // patterns that suggest potential bugs.
-    stage('Static Analysis') {
-      steps {
-        sh './gradlew sonarqube'
-        // wait for sonarqube to finish its analysis
-        sleep 5
-        sh './gradlew checkQualityGate'
-      }
-    }
+  // run the tests which require connection to a
+  // running database.
+  stage('Database Tests') {
+    sh './gradlew integrate'
+    junit 'build/test-results/integrate/*.xml'
+  }
 
+  // These are the Behavior Driven Development (BDD) tests
+  // See the files in src/bdd_test
+  // These tests do not require a running system.
+  stage('Automated Acceptance Tests') {
+      result = sh(script: './gradlew generateCucumberReports', returnStatus: true)
+      results["Acceptance Tests"]["status"] = (result == 0) ? 2 : 1
+      // generate the code coverage report for jacoco
+      sh './gradlew jacocoTestReport'
+      junit 'build/test-results/bdd/*.xml'
+  }
 
-    // Move the binary over to the test environment and
-    // get it running, in preparation for tests that
-    // require a whole system to be running.
-    stage('Deploy to test') {
-      steps {
-      sh './gradlew deployToTestWindowsLocal -Pdeploy_directory=\"/Users/matthew/Downloads/apache-tomcat-9.0.37/webapps\"'
-      // pipenv needs to be installed and on the path for this to work.
-      sh 'PIPENV_IGNORE_VIRTUALENVS=1 pipenv install'
+  // Runs an analysis of the code, looking for any
+  // patterns that suggest potential bugs.
+  stage('Static Analysis') {
+      sh './gradlew sonarqube'
+      // wait for sonarqube to finish its analysis
+      sleep 5
+      sh './gradlew checkQualityGate'
+  }
 
-      // Wait here until the server tells us it's up and listening
-      sh './gradlew waitForHeartBeat'
-      }
-    }
 
+  // Move the binary over to the test environment and
+  // get it running, in preparation for tests that
+  // require a whole system to be running.
+  stage('Deploy to test') {
+    sh './gradlew deployToTestWindowsLocal -Pdeploy_directory=\"/Users/matthew/Downloads/apache-tomcat-9.0.37/webapps\"'
```

Comment:
This is pretty darn specific

Reply:
Hmm, this is necessary to get it working locally, but I'm guessing it's properly configured to work on the boxes without args. Might due to keep a local branch or something, though I hate long lived branches